addons/frag/functions/fnc_dev_switchUnitHandle.sqf

```diff
@@ -0,0 +1,50 @@
+#include "..\script_component.hpp"
+/*
+ * Author: Lambda.Tiger
+ * Handle for debug actions when switching units
+ *
+ * Arguments:
+ * 0: Last unit <OBJECT>
+ * 1: Current unit <OBJECT>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [unit0, player] call ace_frag_fnc_dev_switchUnitHandle;
+ *
+ * Public: No
+ */
+TRACE_1("switchUnit",_this);
+params ["_currentUnit", "_lastUnit"];
+
+if (isNull _currentUnit || {_lastUnit isEqualTo _currentUnit}) exitWith {};
+if (!local _currentUnit) exitWith {
+    [
+        {local _currentUnit},
+        FUNC(dev_switchUnitHandle),
+        _this,
+        5
+    ] call CBA_fnc_waitUntilAndExecute;
+};
+
+private _actionID = missionNamespace getVariable [QGVAR(dev_clearTraceAction), -1];
+if (_actionID > -1 && {!isNull _lastUnit}) then {
+    _lastUnit removeAction _actionID;
+};
+
+_actionID = _currentUnit addAction [
+    "Reset Lines",
+    {
+        remoteExecCall [QFUNC(dev_clearTraces), 2];
```

Comment:
It's fine if it's filtered. The function shouldn't be run on a player in normal gameplay anyways and really only exists in the code loops when debug mode is enabled. Which reminds me, since these functions are *ALWAYS* prepared now, I'll move the dev variables to always be defined in the `preinit`

Reply:
Maybe I misunderstood because the old ACE_FRAG had debug content that was only available through a different compilation and used to visualize the system. Should I include that here?

Reply:
> removed as of [13834d7](https://github.com/acemod/ACE3/commit/13834d7f17bebc2e40cfdc9e192436dffa9d7c4f)

You've just moved, not removed it :P

Make an event that is registered on the server only in postInit (can probably be added in `CBA_settingsInitialized` EH):

```sqf
[
    "CBA_settingsInitialized",
    {
        if (isServer) then {
            call FUNC(initBlackList);
            call FUNC(initMaterialCache);
            
            [QGVAR(dev_clearTraces),LINKFUNC(dev_clearTraces)] call CBA_fnc_addEventHandler;
        };

Reply:
> Maybe I misunderstood because the old ACE_FRAG had debug content that was only available through a different compilation and used to visualize the system. Should I include that here at all?

Usually debug information is enabled through uncommenting `// #define DEBUG_MODE_FULL` and then repacking the PBO.

Reply:
> > removed as of [13834d7](https://github.com/acemod/ACE3/commit/13834d7f17bebc2e40cfdc9e192436dffa9d7c4f)
> 
> You've just moved, not removed it :P
Right, but I don't think it matters

Reply:
I kinda get it thinking about it further. I guess it's easier to avoid having to make sure your test environment is clean or somethign along those lines, sorry for being thick headed about it.